packages/react-vapor/src/components/navigation/pagination/NavigationPagination.tsx

```diff
@@ -39,7 +39,13 @@ export class NavigationPagination extends React.Component<INavigationPaginationP
         }
     };
 
-    componentWillMount() {
+    componentDidUpdate() {
+        if (this.props.onPageClick && this.props.currentPage > this.props.totalPages - 1) {
+            this.props.onPageClick(this.props.totalPages - 1);
+        }
+    }
+
+    componentDidMount() {
         if (this.props.onRender) {
             this.props.onRender();
         }

```

Comment:
I don't know if you want to change it at the same time: 

`this.props.onRender?.()`

Reply:
OH Yes !!